azurelinuxagent/distro/suse/rdma.py

```diff
@@ -0,0 +1,82 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2014 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.4+ and Openssl 1.0+
+#
+
+import glob
+import os
+import azurelinuxagent.logger as logger
+import azurelinuxagent.utils.shellutil as shellutil
+from azurelinuxagent.distro.default.rdma import RDMAHandler
+
+
+class SUSERDMAHandler(RDMAHandler):
+    def install_driver(self):
+        """Install the appropropriate driver package for the RDMA firmware"""
+
+        fw_version = self._RDMAHandler__get_rdma_version()
+        if not fw_version:
+            error_msg = 'Could not determine firmware version. Therefore, '
+            error_msg += 'no driver will be installed.'
+            logger.error(error_msg)
+            return
+        zypper_install = 'zypper -n in'
+        zypper_search = 'zypper se -s'
+        package_name = 'msft-rdma-kmp-default'
+        cmd = '%s %s' % (zypper_search, package_name)
+        status, repo_package_info = shellutil.run_get_output(cmd)
+        for entry in repo_package_info.split('\n'):
+            if package_name in entry:
+                sections = entry.split('|')
+                if len(sections) < 4:
+                    error_msg = 'Unexpected output of "%s" with result "%s"'
+                    logger.error(error_msg % (cmd, entry))
```

Comment:
On 06/10/2016 06:33 PM, Hans Krijger wrote:
> In azurelinuxagent/distro/suse/rdma.py
> <https://github.com/Azure/WALinuxAgent/pull/227#discussion_r66690058>:
> 
>> +        if not fw_version:
>> +            error_msg = 'Could not determine firmware version. Therefore, '
>> +            error_msg += 'no driver will be installed.'
>> +            logger.error(error_msg)
>> +            return
>> +        zypper_install = 'zypper -n in'
>> +        zypper_search = 'zypper se -s'
>> +        package_name = 'msft-rdma-kmp-default'
>> +        cmd = '%s %s' % (zypper_search, package_name)
>> +        status, repo_package_info = shellutil.run_get_output(cmd)
>> +        for entry in repo_package_info.split('\n'):
>> +            if package_name in entry:
>> +                sections = entry.split('|')
>> +                if len(sections) < 4:
>> +                    error_msg = 'Unexpected output of "%s" with result "%s"'
>> +                    logger.error(error_msg % (cmd, entry))
> 
> should we break after this error?

Lets continue in the hopes that another entry may be found.